rcon/squad_automod/automod.py

```diff
@@ -308,12 +290,15 @@ def _do_punitions(
         return
     for aplayer in players:
         try:
-            if method == "punish":
+            if method == "message":
+                audit(config, f"Warnings: {message}")
+                rcon.do_message_player(aplayer.player, aplayer.steam_id_64, message, by=AutomodUsername)
```

Comment:
I'm worried that with all those calls we're going to add quite some latency and we may start to see more false positives if ppl take the officer role while a punition loop is applying. 

We could either use the thread pool I added a while back. or as a simpler solution, start with the punish which should be timely. Then the messages and kicks. What do you think?

Reply:
I think with the threadpool we would be still constrained if the server where the tool is running is not powerful enough to get the work done. Changing the order would probably the best thing (I would also do the kicks before the messages, isn't it? It's pretty destructive as well)